actions/v2/data/server.go

```diff
@@ -0,0 +1,20 @@
+package data
+
+import (
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/rs/zerolog"
+)
+
+// APIData represents server with API endpoints
+type APIData struct {
+	engine engine.ClientInterface
+	logger *zerolog.Logger
+}
+
+// NewAPIData creates a new server with API endpoints
+func NewAPIData(engine engine.ClientInterface, log *zerolog.Logger) APIData {
+	return APIData{
+		engine: engine,
+		logger: log,
```

Comment:
Maybe you could add everywhere in NewAPIXyz something like:
```suggestion
		logger: log.With().Str("api","data").Logger(),
```
Or some other key